lib/src/blaze/block_witness_data.rs

```diff
@@ -510,23 +513,24 @@ impl BlockAndWitnessData {
         [u8; 32]: From<<D as Domain>::ExtractedCommitmentBytes>,
         D::Recipient: crate::wallet::traits::Recipient,
     {
-        // Get the previous block's height, because that block's sapling tree is the tree state at the start
+        // Get the previous block's height, because that block's commitment trees are the states at the start
         // of the requested block.
         let prev_height = { u64::from(height) - 1 };
 
         let (cb, mut tree) = {
+            // In the edge case of a transition to a new network epoch, there is no previous tree.
```

Comment:
Specifically, there's no previous tree for the upgrade in question. At orchard activation height there's no previous orchard tree, but there's still a previous sapling tree.